app/controlplane/internal/data/workflow.go

```diff
@@ -218,6 +218,26 @@ func (r *WorkflowRepo) FindByID(ctx context.Context, id uuid.UUID) (*biz.Workflo
 	return entWFToBizWF(workflow, lastRun)
 }
 
+func (r *WorkflowRepo) FindByName(ctx context.Context, name string) (*biz.Workflow, error) {
+	workflow, err := r.data.db.Workflow.Query().
+		Where(workflow.DeletedAtIsNil(), workflow.Name(name)).
+		WithContract().WithOrganization().
+		Only(ctx)
+	if err != nil && !ent.IsNotFound(err) {
+		return nil, err
+	} else if workflow == nil {
+		return nil, nil
+	}
```

Comment:
you can use the other pattern we are already applying in other places, where the error translation is done in the repository layer:
```suggestion
	if err != nil {
		if ent.IsNotFound(err) {
			return nil, biz.NewErrNotFound("workflow")
		}
		return nil, err
	}
```

Reply:
Ok, I'll also revisit where I taken it from because might be wrong as well.